### DISCUSSION TITLE
Apply DOM changes on new content from overlay

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#636] [Q&A] [answered]
Hi.

I'm trying to use the Bootstrap tooltip on a layer that is loaded from a link. The JS that Bootstrap suggest doesn't find the new content to apply the tooltips.

```
up.on('up:fragment:loaded', (event) => {
    const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]')
    const tooltipList = [...tooltipTriggerList].map(tooltipTriggerEl => new bootstrap.Tooltip(tooltipTriggerEl))
})
``` 

Apparently `document` doesn't have the new fragment. Is that expected or am I doing something wrong?

TIA

**Comment by adam12** [comment, marked as answer]
Try using a compiler instead. I suspect `up:fragment:loaded` is too early.

```js
up.compiler('[data-bs-toggle="tooltip"]', (el) => {
    const tooltipTriggerList = document.querySelectorAll('[data-bs-toggle="tooltip"]')
    const tooltipList = [...tooltipTriggerList].map(tooltipTriggerEl => new bootstrap.Tooltip(tooltipTriggerEl))
})
```

You should clean this up to use the element from the compiler instead of looking over the document again, but hopefully it unblocks you.

**Reply by alexandremjacques** [reply]
Worked. Thanks!
